### PR TITLE
OCPBUGS-1126: [release-4.11] irqbalance: add unit to clear the cpu ban list

### PR DESCRIPTION
**Pull request by marioferh**
Manual cherry pick from https://github.com/openshift/cluster-node-tuning-operator/pull/444

### CODE REVIEW

**Comment by fromanirh** [comment]
interesting, the e2e failure looks legit

**Comment by marioferh** [comment]
I is, maybe bad conflict resolution. Not its done

**Comment by fromanirh** [comment]
/approve
we decided we want this change, so we need it in 4.11

**Review by fromanirh — COMMENTED** [review]
/lgtm
the backport seems correct

File: test/e2e/performanceprofile/functests/1_performance/irqbalance.go [hunk 1]
```diff
@@ -163,6 +167,89 @@ var _ = Describe("[performance] Checking IRQBalance settings", func() {
...
+
+		It("Should store empty cpu mask in the backup file", func() {
+			// crio stores the irqbalance CPU ban list in the backup file once, at startup, if the file doesn't exist.
+			// This _likely_ means the first time the provisioned node boots, and in this case is _likely_ the node
+			// has not any IRQ pinning, thus the saved CPU ban list is the empty list. But we don't control nor declare this state.
+			// It's all best effort.
+
+			nodeIdx := pickNodeIdx(workerRTNodes)
+			node := &workerRTNodes[nodeIdx]
+			By(fmt.Sprintf("verifying worker node %q", node.Name))
```
Comment (shajmakh) [inline test/e2e/performanceprofile/functests/1_performance/irqbalance.go:179]: minor: why not using the targetNode initiated the before-each?

File: test/e2e/performanceprofile/functests/1_performance/irqbalance.go [hunk 2]
```diff
@@ -163,6 +167,89 @@ var _ = Describe("[performance] Checking IRQBalance settings", func() {
...
+			origBannedCPUsFile := "/etc/sysconfig/orig_irq_banned_cpus"
+			By(fmt.Sprintf("Checking content of %q on node %q", origBannedCPUsFile, node.Name))
+			expectFileEmpty(node, origBannedCPUsFile)
+		})
+
+		It("Should DO overwrite the banned CPU set on CRI-O restart", func() {
+
+			nodeIdx := pickNodeIdx(workerRTNodes)
+			node := &workerRTNodes[nodeIdx]
+			By(fmt.Sprintf("verifying worker node %q", node.Name))
```
Comment (shajmakh) [inline test/e2e/performanceprofile/functests/1_performance/irqbalance.go:202]: same here

File: test/e2e/performanceprofile/functests/1_performance/irqbalance.go [hunk 3]
```diff
@@ -163,6 +167,89 @@ var _ = Describe("[performance] Checking IRQBalance settings", func() {
...
+			smpAffinitySet, err := nodes.GetDefaultSmpAffinitySet(node)
+			Expect(err).ToNot(HaveOccurred(), "failed to get default smp affinity")
+
+			By(fmt.Sprintf("Checking the online CPU Set on node %q", node.Name))
+			onlineCPUsSet, err := nodes.GetOnlineCPUsSet(node)
+			Expect(err).ToNot(HaveOccurred(), "failed to get Online CPUs list")
+
+			// expect no irqbalance run in the system already, AKA start from pristine conditions.
+			// This is not an hard requirement, just the easier state to manage and check
+			Expect(smpAffinitySet.Equals(onlineCPUsSet)).To(BeTrue(), "found default_smp_affinity %v, expected %v - IRQBalance already run?", smpAffinitySet, onlineCPUsSet)
```
Comment (shajmakh) [inline test/e2e/performanceprofile/functests/1_performance/irqbalance.go:216]: this section was already checked in the previous It why double-check it if the "It"s run one after the other and there is no change expected in the banned cpus file?

File: test/e2e/performanceprofile/functests/1_performance/irqbalance.go [hunk 4]
```diff
@@ -163,6 +167,89 @@ var _ = Describe("[performance] Checking IRQBalance settings", func() {
 
 			Expect(postRestartBannedCPUs.ToSlice()).To(Equal(postCreateBannedCPUs.ToSlice()), "banned CPUs changed post tuned restart on node %q", postRestartBannedCPUs.ToSlice(), targetNode.Name)
 		})
+
+		It("Should store empty cpu mask in the backup file", func() {
+			// crio stores the irqbalance CPU ban list in the backup file once, at startup, if the file doesn't exist.
+			// This _likely_ means the first time the provisioned node boots, and in this case is _likely_ the node
```
Comment (Tal-or) [inline test/e2e/performanceprofile/functests/1_performance/irqbalance.go:173]: The comment is not so clear
What  \_\_likely\_\_ convention means? I know this term from the C programming world but I don't understand how it fits in here

File: assets/performanceprofile/scripts/clear-irqbalance-banned-cpus.sh [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+[ ! -f ${IRQBALANCE_CONF} ] && exit 0
+
+${SED} -i '/^\s*IRQBALANCE_BANNED_CPUS\b/d' ${IRQBALANCE_CONF} || exit 0
+echo "IRQBALANCE_BANNED_CPUS=" >> ${IRQBALANCE_CONF}
+
+# we now own this configuration. But CRI-O has code to restore the configuration,
+# and until it gains the option to disable this restore flow, we need to make
+# the configuration consistent such as the CRI-O restore will do nothing.
+if [ -n ${CRIO_ORIG_BANNED_CPUS} ] && [ -f ${CRIO_ORIG_BANNED_CPUS} ]; then
+	true > ${CRIO_ORIG_BANNED_CPUS}
```
Comment (Tal-or) [inline assets/performanceprofile/scripts/clear-irqbalance-banned-cpus.sh:20]: This line acts like `touch` is that the intended behavior?

File: test/e2e/performanceprofile/functests/1_performance/irqbalance.go (hunk 4 above)
Comment (fromanirh) [reply test/e2e/performanceprofile/functests/1_performance/irqbalance.go:173]: this is just the word "likely" (in plain english meaning) emphasized in the text :) Is intended to be read as underlined text for emphasis

File: assets/performanceprofile/scripts/clear-irqbalance-banned-cpus.sh (hunk 5 above)
Comment (fromanirh) [reply assets/performanceprofile/scripts/clear-irqbalance-banned-cpus.sh:20]: more than that: it will truncate the destination file to zero size
